### DISCUSSION TITLE
Question for utxos when signing Bitcoin transaction

### DISCUSSION THREAD

**Discussion by SinestCorporation** [trustwallet/wallet-core#2742] [Q&A] [answered]
When signing bitcoin, if I put all available utxos in the addAllUtxo function, it will automatically select as many as I need?

                      val input = Bitcoin.SigningInput.newBuilder().apply {
                            this.amount = satoshiAmount
                            hashType = BitcoinSigHashType.ALL.value()
                            this.toAddress = toAddress
                            changeAddress = address
                            byteFee = fee
                            this.coinType = coinType.value()
                            addAllUtxo(utxoList) <--------------------------------------- here 
                            addPrivateKey(ByteString.copyFrom(key))
                        }

**Comment by hewigovens** [comment, marked as answer]
Yes, it will, we have another plan api to do utxo selection in `AnySigner.plan`

**Reply by SinestCorporation** [reply]
Thanks!
